### PR TITLE
Add As4630 platform support

### PR DESCRIPTION
**Pull request by rubensfig**
This PR aims to add the necessary files to add support for the accton-as4630. More info in the individual commits

### CODE REVIEW

File: conf/machine/accton-as4630-54pe.conf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+MACHINE_FEATURES += "pcbios efi"
+MACHINE_FEATURES_remove += "alsa"
+
+SYSLINUX_OPTS = "serial 1 115200"
+SERIAL_CONSOLE = "115200 ttyS0"
+APPEND += "console=ttyS0,115200 console=tty0"
+
+PLATFORM_VENDOR = "accton"
+BISDN_ONIE_MACHINE = "as4630-54pe"
+BISDN_ONIE_PLATFORM = "x86_64-accton-as7726-r0"
```
Comment (KanjiMonster) [inline conf/machine/accton-as4630-54pe.conf:27]:
```suggestion
BISDN_ONIE_PLATFORM = "x86_64-accton-as4630-54pe-r0"
```

File: recipes-core/platform-as4630-54pe/files/platform-as4630-54pe-init.sh [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+# init PSU-1
+create_i2c_dev as4630_54pe_psu2 0x51 11
+create_i2c_dev ype1200am 0x59 11
+
+for port in {49..53}; do
+	add_port 'optoe2' $port $((port-31))
+done
+for port in {53..55}; do
+	add_port 'optoe1' $port $((port-31))
+done
```
Comment (KanjiMonster) [inline recipes-core/platform-as4630-54pe/files/platform-as4630-54pe-init.sh:48]: You have overlapping ranges here; first ports 49 to 53, then ports 53 to 55.

File: recipes-extended/onl/onl/0001-accton-as4630-54pe-silence-error.patch [hunk 3]
```diff
@@ -0,0 +1,28 @@
+From 9301acb8e29fcc27716cc3183005288f9e23e801 Mon Sep 17 00:00:00 2001
+From: Rubens Figueiredo <rubens.figueiredo@bisdn.de>
+Date: Wed, 10 Nov 2021 11:56:16 +0100
+Subject: [PATCH] accton-as4630-54pe: silence error
+
+x86_64_accton_as4630_54pe/module/src/sysi.c:256:9:
+error: ignoring return value of 'system', declared with attribute
+warn_unused_result [-Werror=unused-result]
+---
```
Comment (KanjiMonster) [inline recipes-extended/onl/onl/0001-accton-as4630-54pe-silence-error.patch:9]: Please sign off the patch as well.

File: recipes-kernel/accton-as4630-54pe-leds-mod/accton-as4630-54pe-leds-mod.bb [hunk 4]
```diff
@@ -0,0 +1,17 @@
+SUMMARY = "This module supports cpld that read and write"
```
Comment (KanjiMonster) [inline recipes-kernel/accton-as4630-54pe-leds-mod/accton-as4630-54pe-leds-mod.bb:1]: Different module.

File: recipes-kernel/accton-as4630-54pe-psu-mod/accton-as4630-54pe-psu-mod.bb [hunk 5]
```diff
@@ -0,0 +1,17 @@
+SUMMARY = "This module supports cpld that read and write"
```
Comment (KanjiMonster) [inline recipes-kernel/accton-as4630-54pe-psu-mod/accton-as4630-54pe-psu-mod.bb:1]: Different module.

File: recipes-kernel/accton-as4630-54pe-cpld-mod/accton-as4630-54pe-cpld-mod.bb [hunk 6]
```diff
@@ -0,0 +1,15 @@
+SUMMARY = "This module supports cpld that read and write"
```
Comment (KanjiMonster) [inline recipes-kernel/accton-as4630-54pe-cpld-mod/accton-as4630-54pe-cpld-mod.bb:1]: Maybe use a better summary, e.g. the first sentence in the comment block in the c file.

File: recipes-kernel/accton-as4630-54pe-cpld-mod/accton-as4630-54pe-cpld-mod.bb [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+LICENSE = "GPLv2"
+LIC_FILES_CHKSUM = "file://COPYING;md5=12f884d2ae1ff87c09e5b7ccc2c4ca7e"
+
+inherit module
+
+SRC_URI = " \
+	  file://Makefile \
+          file://x86-64-accton-as4630-54pe-cpld.c \
+	  file://COPYING \
```
Comment (KanjiMonster) [inline recipes-kernel/accton-as4630-54pe-cpld-mod/accton-as4630-54pe-cpld-mod.bb:10]: Please align the lines and check the other recipes for similar issues.

**Review by KanjiMonster — CHANGES_REQUESTED** [review]
In general the direction is fine.

Please use the same commit message prefix for all commits. Make sure the machine definition is last.

Also there are at least 4 open PRs at https://github.com/opencomputeproject/OpenNetworkLinux/pulls?q=is%3Apr+is%3Aopen+as4630 that look relevant, which we should have as well.

Ideally add each as an individual commit.

**Review by KanjiMonster — CHANGES_REQUESTED** [review]
Please add a bit more description of the as4630 in the PR.

Please apply all patches in onl that touch kernel drivers also to our copies.

Please make "onl: " the first prefix in commits touching (only) onl. If the commit also touches other parts in the tree, it's fine to not have that prefix.

File: recipes-extended/onl/onl/0001-as4630-54pe-Fix-fan-LED-always-showing-fan-is-turned.patch [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+Signed-off-by: Jostar Yang <jostar_yang@accton.com.tw>
+---
+ .../builds/x86-64-accton-as4630-54pe-leds.c   | 46 ++++---------------
+ .../module/src/ledi.c                         | 19 ++------
+ 2 files changed, 13 insertions(+), 52 deletions(-)
+
+diff --git a/packages/platforms/accton/x86-64/as4630-54pe/modules/builds/x86-64-accton-as4630-54pe-leds.c b/packages/platforms/accton/x86-64/as4630-54pe/modules/builds/x86-64-accton-as4630-54pe-leds.c
+index 04caf13e..ff5d01c5 100755
+--- a/packages/platforms/accton/x86-64/as4630-54pe/modules/builds/x86-64-accton-as4630-54pe-leds.c
++++ b/packages/platforms/accton/x86-64/as4630-54pe/modules/builds/x86-64-accton-as4630-54pe-leds.c
```
[the remark this answers is not in the document]
Comment (rubensfig) [reply recipes-extended/onl/onl/0001-as4630-54pe-Fix-fan-LED-always-showing-fan-is-turned.patch:18]: I applied the patch in my personal tree, so the state reflected here already has the state attached.

But I agree, its clear to get the patch individually and apply it in the recipe

**Review by KanjiMonster — CHANGES_REQUESTED** [review]
Looks good in general, apart from the merge conflict remnants in https://github.com/bisdn/meta-open-network-linux/pull/16/commits/269ac3c623e66a5ddeaf7434ad366f3d0231288e in the commit message.

**Review by KanjiMonster — APPROVED** [review]

**Pull request merged**